_layouts/ndt-test.html

```diff
@@ -2,33 +2,116 @@
 <!DOCTYPE html>
 <html>
 <head>
-	<meta charset="utf8">
-	<title>{{ page.title }}</title>
-	<style>
-		#ndt-iframe {
-			width: 720px;
-			height: 480px;
-			border: none;
-		}
-	</style>
-</head>
-<body>
-<script>
-var	mlabNsUrl = 'https://mlab-ns.appspot.com/ndt?format=json';
-var xhr = new XMLHttpRequest();
-xhr.onreadystatechange = function () {
-  if (xhr.readyState === 4) {
-    if (xhr.status === 200) {
-      var response = JSON.parse(xhr.responseText);
-      var ndtIframe = document.createElement('iframe');
-      ndtIframe.src = response.url;
-      ndtIframe.setAttribute('id', 'ndt-iframe');
-      document.body.appendChild(ndtIframe);
+    <meta charset="utf8">
+    <title>{{ page.title }}</title>
+    <link href="{{ site.baseurl }}/css/ndt-test.css" rel="stylesheet">
+    <!--[if IE]>
+    <style>
+    @font-face {
+        font-family: "League Gothic";
+        src: url("../fonts/league-gothic/league_gothic.eot");
     }
-  }
-};
-xhr.open("GET", mlabNsUrl, true);
-xhr.send();
-</script>
+    </style>
+    <![endif]-->
+    <script>
+        // The gauge.min.js in the ndt submodule wants the font digital-7-mono,
+        // and by default looks for it in the /fonts directory.  Setting this
+        // variable tells it where to look.
+        var CANV_GAUGE_FONTS_PATH = "../fonts/digital-7-mono";
+    </script>
+    <script src="{{ site.baseurl }}/ndt/HTML5-frontend/jquery-1.12.1.min.js"></script>
+    <script src="{{ site.baseurl }}/ndt/HTML5-frontend/gauge.min.js"></script>
+    <script src="{{ site.baseurl }}/ndt/HTML5-frontend/script.js"></script>
+    <script src="{{ site.baseurl }}/ndt/HTML5-frontend/ndt-browser-client.js"></script>
+    <script src="{{ site.baseurl }}/ndt/HTML5-frontend/ndt-wrapper.js"></script>
+    <script>
+        // Lookup a suitable NDT server using mlab-ns. The variable ndtServer is
+        // used by script.js in the ndt submodule
+        var ndtServer;
+        var mlabNsService = 'https:' == location.protocol ? 'ndt_ssl' : 'ndt';
+        var mlabNsUrl = 'https://mlab-ns.appspot.com/';
+        $.ajax({
```

Comment:
I don't think so.  To avoid this is why I've set "async: false" in the $.ajax() request. My understanding is that this should block the entire page until the call finishes.  I think the reason you are seeing that error is because the front-end code changes, but the ndt submodule was still pointing to the outdated generic-js branch of the m-lab/ndt repo.  I've updated the submodule and expect it to work after I push the commits to the PR in a minute.